service/history/workflow/errors.go

```diff
@@ -34,7 +34,7 @@ var (
 	// ErrActivityTaskNotFound is the error to indicate activity task could be duplicate and activity already completed
 	ErrActivityTaskNotFound = &types.EntityNotExistsError{Message: "activity task not found"}
 	// ErrAlreadyCompleted is the error to indicate workflow execution already completed
-	ErrAlreadyCompleted = &types.EntityNotExistsError{Message: "workflow execution already completed"}
+	ErrAlreadyCompleted = &types.WorkflowExecutionAlreadyCompletedError{Message: "workflow execution already completed"}
```

Comment:
This error is used by many other APIs (for example RecordChildExecutionCompleted) besides the three that have their IDL definition updated. We will need to add already completed error to the IDL definition for all those APIs I think.

Reply:
Yeah I realized this while looking at the integration test failures. Right now the differentiation is based on if the request was signal/terminate/cancel type or not. I will probably update this diff with the idl repo later. Let me know if you have any other ideas. I updated the pr to fix the integration tests though.